src/platforms/salesforce/schemas.ts

```diff
@@ -203,16 +226,415 @@ export const salesforceOpportunityUpdate = validators.object({
-  LastModifiedDate: validators.date(),
-  CreatedById: z.string(),
+  Task: z
+    .object({
+      ActivityDate: validators.date(),
+      Description: z.string(),
+      OwnerId: z.string(),
+      Priority: z.string(),
+      Status: z.string(),
+      Subject: z.string(),
+      CallDisposition: z.string(),
+      CallType: z.enum(SALESFORCE_CALL_TYPES),
+      TaskSubtype: z.literal('Call'),
+      Type: z.literal('Call'),
+    })
+    .partial(),
 });
 
+// -
+// Event
+// -
+export const salesforceEvent = validators
+  .object({
+    Id: z.string(),
+    OwnerId: z.string(),
+    CreatedDate: validators.date(),
+    LastModifiedDate: validators.date(),
+    Subject: z.string(),
+    Type: z.string(),
+    Description: z.string(),
+    StartDateTime: validators.date(),
+    EndDateTime: validators.date(),
+    Location: z.string(),
+    IsAllDayEvent: z.boolean(),
+    ActivityDate: validators.date(),
+    ActivityDateTime: validators.date(),
+    AcceptedEventInviteeIds: z.array(z.string()),
+    DeclinedEventInviteeIds: z.array(z.string()),
+    UndecidedEventInviteeIds: z.array(z.string()),
+    Who: z.object({
+      Id: z.string(),
+      Type: z.string(),
+    }),
+    What: z.object({
+      Id: z.string(),
+      Type: z.string(),
+    }),
+  })
+  .partial()
+  .required(requiredFields);
+export const salesforceEventRelationalSelect = {
+  Account: 'What.Id',
+  Opportunity: 'What.Id',
+  Contact: 'Who.Id',
+  Lead: 'Who.Id',
+};
+
+export const salesforceEventCreate = validators.object({
+  Event: z
+    .object({
+      Subject: z.string(),
+      Type: z.string(),
+      Description: z.string(),
+      StartDateTime: validators.date(),
+      EndDateTime: validators.date(),
+      Location: z.string(),
+      IsAllDayEvent: z.boolean(),
+      WhoId: z.string(),
+      WhatId: z.string(),
+    })
+    .partial(),
+});
+
+export const salesforceEventCreateResponse = validators.object({
+  id: z.string(),
+});
+
+export const salesforceEventUpdate = validators.object({
+  Id: z.string(),
+  Event: z
+    .object({
+      Subject: z.string(),
+      Type: z.string(),
+      Description: z.string(),
+      StartDateTime: validators.date(),
+      EndDateTime: validators.date(),
+      Location: z.string(),
+      IsAllDayEvent: z.boolean(),
+    })
+    .partial(),
+});
+
+// -
+// EventRelation
+// -
+export const salesforceEventRelation = validators
+  .object({
+    Id: z.string(),
+    CreatedDate: validators.date(),
+    LastModifiedDate: validators.date(),
+    EventId: z.string(),
+    RelationId: z.string(),
+    Status: z.string(),
+    IsInvitee: z.boolean(),
+  })
+  .partial()
+  .required(requiredFields);
+
+export const salesforceEventRelationCreate = validators.object({
+  EventRelation: z
+    .object({
+      EventId: z.string(),
+      RelationId: z.string(),
+      Status: z.string(),
+    })
+    .partial(),
+});
+
+export const salesforceEventRelationCreateResponse = validators.object({
+  id: z.string(),
```

Comment:
Making sure this is supposed to be lower case too? If so I hate salesforce

Reply:
Yeah haha, the REST API returns lower-case `id`, as opposed to using SOQL queries